legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/test/ModelStoreTestConnectionFactory.java

```diff
@@ -12,76 +12,81 @@
 import org.finos.legend.engine.protocol.pure.v1.model.data.*;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.connection.Connection;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.data.DataElement;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.Store;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.modelToModel.ModelStore;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.modelToModel.connection.JsonModelConnection;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.modelToModel.connection.XmlModelConnection;
 import org.finos.legend.engine.protocol.pure.v1.model.valueSpecification.ValueSpecification;
 import org.finos.legend.engine.protocol.pure.v1.model.valueSpecification.raw.PackageableElementPtr;
+import org.finos.legend.engine.shared.core.url.InputStreamProvider;
+import org.finos.legend.engine.shared.core.url.StreamProviderHolder;
+
+import java.io.ByteArrayInputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
 
+// TO BE MOVED TO MODEL STORE MODULE ONCE CREATED
 public class ModelStoreTestConnectionFactory implements ConnectionFactoryExtension
 {
-    public static String MODEL_STORE = "ModelStore";
+    public static final String MODEL_STORE = "ModelStore";
     //TODO: after refactor use the already present variables
     //currently creates a circular dependency, hence cannot use the already present variables
-    private static String DATA_PROTOCOL_NAME = "data";
-    private static String APPLICATION_JSON = "application/json";
-    private static String APPLICATION_XML = "application/xml";
+    private static final String DATA_PROTOCOL_NAME = "data";
+    private static final String APPLICATION_JSON = "application/json";
+    private static final String APPLICATION_XML = "application/xml";
+
+    private static JsonModelConnection jsonModelConnection = new JsonModelConnection();
+    private static XmlModelConnection xmlModelConnection = new XmlModelConnection();
+
 
-    private Connection resolveExternalFormatData(ExternalFormatData externalFormatData, String _class, boolean useDefaultExecutor)
+    private Pair<Connection, List<Closeable>> resolveExternalFormatData(ExternalFormatData externalFormatData, String _class)
     {
+        Closeable closeable = new Closeable()
+        {
+            @Override
+            public void close() throws IOException
+            {
+                StreamProviderHolder.streamProviderThreadLocal.remove();
+            }
+        };
         if (APPLICATION_JSON.equals(externalFormatData.contentType))
         {
-            JsonModelConnection jsonModelConnection = new JsonModelConnection();
             jsonModelConnection._class = _class;
```

Comment:
Thinking about it, this gives a bit of wrong impression that we are populating connections again and again. It is static so setting it once is essentially what is needed so we should move it to static block to avoid confusion